src/YesSql.Core/Services/DefaultQuery.cs

```diff
@@ -784,7 +784,15 @@ public async Task<int> CountAsync()
             var transaction = await _session.DemandAsync();
 
             var localBuilder = _queryState._sqlBuilder.Clone();
-            localBuilder.Selector("count(*)");
+
+            if (localBuilder.HasJoin)
+            {
+                localBuilder.Selector($"count(distinct {_queryState._sqlBuilder.FormatColumn(_queryState._documentTable, "id")})");
```

Comment:
You are missing the table prefix.

Reply:
I don't think it's missing, it seems to be working fine